kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -145,6 +163,35 @@ public void writeRows(PartitionedTableId table,
         mostRecentException);
   }
 
+  /**
+   * Decide whether the failure is a partial failure or complete failure
+   * @param numOfFailedRows Number of rows that failed
+   * @param numOfAllRows Number of rows to insert
+   * @return isPartialFailure
+   */
+  private boolean isPartialFailure(int numOfFailedRows, int numOfAllRows) {
```

Comment:
I'd prefer this take in the list of attempted rows and list of failed rows, in case we want to make the check more nuanced later.

Reply:
got it